### PR TITLE
Feature/#13 localstorage

### LINKED ISSUE #13: To do list should be stored in `localStorage`
Whenever you add to the list, remove from the list, this should be reflected in `localStorage`

### PR DESCRIPTION
**Pull request by dennisxu54**
closes #13

### CODE REVIEW

File: src/pages/home-page/index.tsx [hunk 1]
```diff
@@ -26,6 +26,7 @@ function HomePage() {
     // 'select'ing is a way to get stuff from the redux store
     // useDispatch and useSelector
     dispatch(addItemToList(item));
+    localStorage.setItem(id.toString(), JSON.stringify(item))
```
Comment (Chidoge) [inline src/pages/home-page/index.tsx:29]: You should not be writing each task to a separate key, instead a better approach would be to set the entire list under a storage key named `list` and stringifying the array and passing that in.

File: src/components/task-card/TaskCard.tsx [hunk 2]
```diff
@@ -17,6 +17,7 @@ const TaskCard = ({ item }: Props) => {
 
   function deleteItem() {
     dispatch(removeItemToList(id));
+    localStorage.removeItem(id.toString())
```
Comment (Chidoge) [inline src/components/task-card/TaskCard.tsx:20]: Read comment about what should be passed in

**Review by Chidoge — CHANGES_REQUESTED** [review]

File: src/pages/task-page/index.tsx [hunk 3]
```diff
@@ -7,6 +7,7 @@ import { ToDoItem } from "../../interfaces/interfaces";
 
 function TaskPage() {
   const list = useSelector((state: RootState) => state.list.list);
+  localStorage.setItem("List", JSON.stringify(list))
```
Comment (Chidoge) [inline src/pages/task-page/index.tsx:10]: Nop, you only want to set the list whenever you add an item to the list, not here

File: src/pages/home-page/index.tsx [hunk 4]
```diff
@@ -32,6 +34,10 @@ function HomePage() {
     setPriority("LOW");
   };
 
+  useEffect(() => {
```
Comment (Chidoge) [inline src/pages/home-page/index.tsx:37]: Same deal here

**Review by Chidoge — CHANGES_REQUESTED** [review]

File: src/pages/home-page/index.tsx [hunk 5]
```diff
@@ -26,6 +26,9 @@ function HomePage() {
     // 'select'ing is a way to get stuff from the redux store
     // useDispatch and useSelector
     dispatch(addItemToList(item));
+    const dataItem = localStorage.getItem('list')
```
Comment (Chidoge) [inline src/pages/home-page/index.tsx:29]: Almost there, I want you to be to updating `localStorage` within the `addItemToList` function

File: src/components/task-card/TaskCard.tsx [hunk 6]
```diff
@@ -17,6 +17,9 @@ const TaskCard = ({ item }: Props) => {
 
   function deleteItem() {
     dispatch(removeItemToList(id));
+    const dataItem = localStorage.getItem('list')
```
Comment (Chidoge) [inline src/components/task-card/TaskCard.tsx:20]: Ditto here

**Review by Chidoge — CHANGES_REQUESTED** [review]

**Comment by dennisxu54** [comment]
image

image

Add three entries
image

image

image

image

image

page refresh
image

image

File: src/store/store.ts [hunk 7]
```diff
@@ -7,7 +7,15 @@ export const initialiseStore = () => {
   // the combineReducers function takes an object,
   // the key is the name we register the reducer under
 
-  return createStore(combineReducers({ list: listReducer }));
+  let preLoadedState
```
Comment (Chidoge) [inline src/store/store.ts:10]: Have you checked that this doesn't cause a crash/error when your `localStorage` is empty?

**Comment by dennisxu54** [comment]
image

image

**Comment by Chidoge** [comment]
LGTM

**Review by Chidoge — APPROVED** [review]

**Pull request merged**